pkg/cmd/plan.go

```diff
@@ -0,0 +1,276 @@
+					Value: "github.com/your/module/name",
+				},
+				&cli.StringFlag{
+					Name:     "plan",
+					Aliases:  []string{"p"},
+					Usage:    "specifies the name of the plan to create",
+					Required: true,
+				},
+			},
+			Action: createCommand,
+		},
+		&cli.Command{
+			Name:  "import",
+			Usage: "import a plan from the local filesystem or git repository",
+			Flags: []cli.Flag{
+				&cli.StringFlag{
+					Name:     "source",
+					Usage:    "specifies the source of the plan to be imported, can be git or local.",
+					Required: true,
+				},
+				&cli.BoolFlag{
+					Name:     "git",
+					Usage:    "use git to import (default: false)",
+					Required: false,
+					Value:    false,
+				},
+				&cli.StringFlag{
+					Name:     "name",
+					Usage:    "override the name of the plan directory (default: automatic)",
+					Required: false,
+				},
+			},
+			Action: importCommand,
+		},
+		&cli.Command{
+			Name:  "rm",
+			Usage: "remove a plan directory from $TESTGROUND_HOME",
+			Flags: []cli.Flag{
+				&cli.BoolFlag{
+					Name:     "yes",
+					Usage:    "confirm removal (without this, the command does nothing)",
+					Required: false,
+					Value:    false,
+				},
+			},
+			Action: rmCommand,
+		},
+		&cli.Command{
+			Name:   "list",
+			Usage:  "enumerate all test cases known to the client",
+			Action: listCommand,
+		},
+	},
+}
+
+// These are the variables used for executing templates during `testground plan create`
+// TODO (cory) Make these specific to the target lang?
+type templateVars struct {
+	Name   string
+	Module string
+}
+
+func createCommand(c *cli.Context) error {
+	if c.Args().Len() != 1 {
+		return errors.New("this command requires one argument -- specify the plan name")
+	}
+
+	cfg := &config.EnvConfig{}
+	if err := cfg.Load(); err != nil {
+		return err
+	}
+
+	plan_name := c.String("plan")
+	target_lang := c.String("target")
+	remote := c.String("remote")
+	module := c.String("module")
+
+	pdir := filepath.Join(cfg.Dirs().Plans(), plan_name)
+	repo, err := git.PlainInit(pdir, false)
+	if err != nil {
+		return err
+	}
+
+	if remote != "" {
+		_, err := repo.CreateRemote(&gitcfg.RemoteConfig{
+			Name: "origin",
+			URLs: []string{remote},
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Get file templates for the supplied target lang
+	asset_path := fmt.Sprintf("/%s-templates", target_lang)
+	var tset ttmpl.TemplateSet
+	err = ttmpl.Fill(asset_path, &tset)
+	if err != nil {
+		return err
+	}
+
+	if tset == nil {
+		return fmt.Errorf("unknown language target %s", target_lang)
+	}
+
+	for _, ts := range tset {
+		tmpl, err := template.New(ts.Filename).Parse(ts.Template)
+		if err != nil {
+			return err
+		}
+		f, err := os.Create(filepath.Join(pdir, tmpl.Name()))
+		if err != nil {
+			return err
+		}
+		err = tmpl.Execute(f, templateVars{Name: plan_name, Module: module})
+		if err != nil {
+			return err
+		}
+		f.Close()
+	}
+	return nil
+}
+
+func importCommand(c *cli.Context) error {
```

Comment:
Could we output a friendly message saying something like: "plan imported at: <full_path>; reference it in commands using: <relative_path>`

```
⟩ ./testground plan import --git --source https://github.com/libp2p/test-plans.git
Apr 30 13:28:16.294357	INFO	using home directory: /Users/raul/testground
Apr 30 13:28:16.294430	WARN	no .env.toml found at /Users/raul/testground/.env.toml; system may be misconfigured
Enumerating objects: 54, done.
Counting objects: 100% (54/54), done.
Compressing objects: 100% (41/41), done.
Total 54 (delta 16), reused 36 (delta 11), pack-reused 0
⟩
```

Reply:
I've set this up to print the imported plans using the same format as `plan list`